interfaces/builtin/common.go

```diff
@@ -85,13 +85,7 @@ func (iface *commonInterface) StaticInfo() interfaces.StaticInfo {
 }
 
 // BeforePrepareSlot checks and possibly modifies a slot.
-//
-// If the reservedForOS flag is set then only slots on core snap
-// are allowed.
 func (iface *commonInterface) BeforePrepareSlot(slot *snap.SlotInfo) error {
-	if iface.reservedForOS {
-		return sanitizeSlotReservedForOS(iface, slot)
-	}
```

Comment:
given that this method is optional if I remember correctly, this can probably be just fully removed, and so all the corresponding tests. Should probably be done in a follow up

Reply:
Good point, indeed it's optional. Will do in a followup, thanks.